### ISSUE TITLE
Expose a `batch` method on CachingSession

### ISSUE THREAD

**Issue by colin-grapl** [scylladb/scylla-rust-driver#469]
Currently there doesn't seem to be any support for batch operations on CachingSession.

As `Session` exposes a `.batch`, which takes a Batch, which itself takes PreparedStatement's, it'd be nice to have that natively supported.

**Comment by colin-grapl** [comment]
https://github.com/colin-grapl/scylla-rust-driver/blob/main/scylla/src/transport/caching_session.rs#L97

Wrote this up pretty quickly. I think it gets the idea across. I may have botched the error handling.

**Comment by cvybhu** [comment]
Thank you for the suggestion and code.

It turns out that `CachingSession` doesn't need to handle repreparing statements.
Once #473 is merged it should be easy to add `CachingSession::batch` as well.

One improvement I would do is to do the `add_prepared_statement`s in parallel using something like `join_all`.

**Comment by colin-grapl** [comment]
Ah, that's awesome. That was by far the most confusing part of this. I agree entirely re: using join_all to do this concurrently.

I'll update the PR to:
1. Remove the reprep code (yay)
2. Optimize this a bit

**Comment by colin-grapl** [comment]
^ Done, feel free to grab that code and do whatever you'd like. I don't really want to go through a PR process myself since I have a workaround in my own code.
